tests/Feature/ApiTest.php

```diff
@@ -0,0 +1,512 @@
+            'member_id' => $m_id,
+            'role' => 'dev'
+            ]);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Assign Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateMember
+     * @depends testCreateProject
+     */
+    function unassign($m_id, $p_id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('DELETE', '/api/project/detach',['project_id' => $p_id,'member_id' => $m_id]);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Unassign Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateMember
+     */
+    function deleteMember($id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('DELETE', '/api/members/' . $id);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Member Deleted Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateProject
+     */
+    function testDeleteProject($id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('DELETE', '/api/projects/' . $id);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Project Deleted Successfully.'
+            ]);
+    }
+
+    function testUploadFile()
+    {
+        $name = time();
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/members', [
+            'name' => 'upload' . $name ,
+            'information' => '1234567891',
+            'dob' => '1992-09-27',
+            'position' => 'intern',
+            'phone' => '0965841492',
+            'gender' => 2,
+            'avatar' =>  new UploadedFile('/home/sunshine/Downloads/thank_you_3.jpg', 'thank_you_3.jpg', "image/jpg", 70, null, true)
+            ]);
+        $response->assertStatus(200)
+        ->assertJson(['status' => true, 'message' => 'Member Added Successfully.']);
+    }
+    function testFailUploadFileByFileType()
+    {
+        $name = time();
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/members', [
+            'name' => 'upload' . $name ,
+            'information' => '1234567891',
+            'dob' => '1992-09-27',
+            'position' => 'intern',
+            'phone' => '0965841492',
+            'gender' => 2,
+            'avatar' =>  new UploadedFile('/home/sunshine/Downloads/53604.pdf', '53604.pdf', "application/pdf", 70, null, true)
+            ]);
+        $response->assertStatus(422)
+        ->assertJson(["message"=>"The given data was invalid.","errors"=>["avatar"=>["The file must have an extendsion jpg, png, gif"]]]);
+    }
+}
```

Comment:
Please add a test case or uploading a file with the wrong contents, but the right extension (for example rename a word document to .png) and test that it fails to upload.

Reply:
i have do it